SDL/PixelTriplet.cu

```diff
@@ -433,78 +442,128 @@ __device__ bool SDL::passPT3RZChiSquaredCuts(struct modules& modulesInGPU, uint1
     {
-        return rzChiSquared < 74.19805f;
+        return rzChiSquared < 11.7439f; //CMSSW 12 geo, add charge
     }
     else if(layer1 == 7 and layer2 == 8 and layer3 == 14)
-    {
-        return rzChiSquared < 97.9479f;
+    {   
+        return rzChiSquared < 16.836f; //CMSSW 12 geo, add charge
     }
 
     else if(layer1 == 1 and layer2 == 2 and layer3 == 3)
     {
-        return rzChiSquared < 451.1407f;
+        return rzChiSquared < 9.3866f; //CMSSW 12 geo, add charge
     }
     else if(layer1 == 1 and layer2 == 2 and layer3 == 7)
     {
-        return rzChiSquared < 595.546f;
+        return rzChiSquared < 42.8617f; //CMSSW 12 geo, add charge
     }
 
     else if(layer1 == 1 and layer2 == 7 and layer3 == 8)
     {
-        return rzChiSquared < 518.339f;
+        return rzChiSquared < 8.3363f; //CMSSW 12 geo, add charge
     }
 
     else if(layer1 == 2 and layer2 == 3 and layer3 == 7)
     {
-        return rzChiSquared < 684.253f;
+        return rzChiSquared < 39.2367f; //CMSSW 12 geo, add charge
     }
     else if(layer1 == 2 and layer2 == 3 and layer3 == 12)
     {
-        return rzChiSquared < 684.253f;
+        return rzChiSquared < 51.76f; //CMSSW 12 geo, add charge
     }
+    
     else if(layer1 == 2 and layer2 == 3 and layer3 == 4)
     {
-        return rzChiSquared  < 392.654f;
+        return rzChiSquared < 38.7867f; //CMSSW 12 geo, add charge
     }
 
     else if(layer1 == 2 and layer2 == 7 and layer3 == 8)
     {
-        return rzChiSquared < 518.339f;
+        return rzChiSquared < 342.315f; //CMSSW 12 geo, add charge
     }
     else if(layer1 == 2 and layer2 == 7 and layer3 == 13)
     {
-        return rzChiSquared < 518.339f;
+        return rzChiSquared < 17.2667f; //CMSSW 12 geo, add charge
     }
 
     //default - category not found!
     return true;
 }
 
-__device__ float SDL::computePT3RZChiSquared(struct modules& modulesInGPU, uint16_t* lowerModuleIndices, float* rtPix, float* zPix, float* rts, float* zs)
+__device__ float SDL::computePT3RZChiSquared(struct modules& modulesInGPU, uint16_t* lowerModuleIndices, float* rtPix, float* xPix, float* yPix, float* zPix, float* rts, float* xs, float* ys, float* zs, float pixelSegmentPt, float pixelSegmentPx, float pixelSegmentPy, float pixelSegmentPz, int pixelSegmentCharge, float pixelG, float pixelF, float pixelRadiusPCA)
 { 
     float slope = (zPix[1] - zPix[0])/(rtPix[1] - rtPix[0]);
     float residual = 0;
     float error = 0;
     //hardcoded array indices!!!
     float RMSE = 0;
+
+    float Pt=pixelSegmentPt, Px=pixelSegmentPx, Py=pixelSegmentPy, Pz=pixelSegmentPz;
+    int charge=pixelSegmentCharge;
+    float CenterX=pixelG/100, CenterY=pixelF/100, CircleRadius=pixelRadiusPCA/100; // for calculating initial state corrections
+    float x0 = xPix[0]/100,x1 = xPix[1]/100;
+    float y0 = yPix[0]/100,y1 = yPix[1]/100;
+    float z0 = zPix[0]/100,z1 = zPix[1]/100;
+    float r0 = rtPix[0]/100,r1 = rtPix[1]/100;
```

Comment:
I couldn't find where we use `CenterX`, `CenterY`, `CircleRadius`, `x0`, `y0`, `z0` and `r0`. Are they used somewhere and I missed it?

Reply:
they are no longer used, let me remove those